posthog/api/event.py

```diff
@@ -100,13 +100,22 @@ class EventViewSet(StructuredViewSetMixin, viewsets.ModelViewSet):
 
     def get_queryset(self):
         queryset = cast(EventManager, super().get_queryset()).add_person_id(self.team_id)
-
         if self.action == "list" or self.action == "sessions" or self.action == "actions":
             queryset = self._filter_request(self.request, queryset)
 
         order_by_param = self.request.GET.get("orderBy")
         order_by = ["-timestamp"] if not order_by_param else list(json.loads(order_by_param))
-        return queryset.order_by(*order_by)
+        queryset = queryset.order_by(*order_by)
+        limit_raw = self.request.GET.get("limit")
+        limit: Optional[int]
+        if limit_raw:
+            try:
+                limit = int(limit_raw)
+                valid_ids = queryset.values_list("pk", flat=True)[:limit]
```

Comment:
Not sure about this, what's the goal here? Why don't you use DRF's built in [pagination](https://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination)? I'm not sure if the lazy querying will be smart enough to limit at the DB level instead of fetching everything and slicing in Python. It's also more code to maintain and understand. I'm also not sure how this plays out with CH. Maybe worth asking @mariusandra for input.

Reply:
I'm a little confused where I can add in the pagination, it looks like we just slice things off here? https://github.com/PostHog/posthog/blob/master/posthog/api/event.py#L197

Reply:
nvm . i got it 😸